onnxruntime/core/providers/cpu/tensor/quantize_linear.cc

```diff
@@ -26,43 +26,48 @@ ONNX_CPU_OPERATOR_TYPED_KERNEL(
 template <typename T>
 // formula is Y = (X - ZeroPoint) * Scale
 Status DequantizeLinear<T>::Compute(OpKernelContext* ctx) const {
```

Comment:
Your change fixes DequantizeLinear for CPU, but what about CUDA EP? QuantizeLinear also has an optional zero point tensor and should be fixed up too.

Reply:
There is a bug in onnx type inference to support QuantizeLinear with optional zero point. Will fix that later.